File: back/src/main/java/com/springboot/security/dto/NewUser.java
package com.springboot.security.dto;

import java.util.HashSet;
import java.util.Set;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class NewUser {
	
	@NotBlank
	private String name;
	@NotBlank
	private String userName;
	@Email
	private String email;
	@NotBlank
	private String password;

	private Set<String> roles = new HashSet<>();

}

File: admin/src/app/models/new-user.ts
export class NewUser {
    name: string;
    userName: string;
    email: string;
    password: string;
    
    constructor(name: string, userName: string, email: string, password: string) {
        this.name = name;
        this.userName = userName;
        this.email = email;
        this.password = password;
    }
}
